wordpress-mobile/WordPress-Android
# Update Automattic-Tracks to 3.0.0 and address breaking changes

This PR updates [Automattic-Tracks-Android](https://github.com/Automattic/Automattic-Tracks-Android) to `3.0.0` and addresses the breaking changes by attempting to keep the existing behavior unchanged.

Note that the PR is targeting `deps/main-batch-androidx-compose-kotlin` branch - from https://github.com/wordpress-mobile/WordPress-Android/pull/18364 - because [Tracks library is using Kotlin version `1.8.21` ](https://github.com/Automattic/Automattic-Tracks-Android/blob/trunk/settings.gradle#L4) and `trunk` is at `1.6.10`. Luckily, @ParaskP7 already [did the work to update the Kotlin version to `1.8.21`](https://github.com/wordpress-mobile/WordPress-Android/pull/18364/files#diff-7f825392aa37acd1cee0c2e7b9bb7366ad6eac64f3e6cdd816e156bcb69d30deR2) in https://github.com/wordpress-mobile/WordPress-Android/pull/18364. ❤️ 

As far as I can tell, there are 5 breaking changes from `2.2.0` to `3.0.0`:

1. `CrashLoggingProvider` now requires an `Application` instead of `Context`.
2. `CrashLoggingProvider` now has a new `CoroutineScope` parameter - I used `@Named(APPLICATION_SCOPE)` for this.
3. `CrashLoggingDataProvider` now has a `val user: Flow<CrashLoggingUser?>` property that needs to be overridden instead of `fun userProvider(): CrashLoggingUser?`. I wrapped the current value with `MutableStateFlow` which is what [WCAndroid is using](https://github.com/woocommerce/woocommerce-android/blob/13.5/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt#L45). I originally wrapped it with `flow {}` however when I tested the implementation the user information wasn't available in Sentry. After I made the change in d7b97dfdf7d2e161dbbde04e4b8bbb2686feb0e6, the information showed up as expected.
4. `CrashLoggingDataProvider.applicationContextProvider` now requires a `Flow` - I wrapped the current value with  `MutableStateFlow` for consistency but I think wrapping it with `flow {}` and emitting the single value should work as well.
5. `CrashLoggingDataProvider` now has a `val performanceMonitoringConfig: PerformanceMonitoringConfig` property that needs to be overridden. I've used `PerformanceMonitoringConfig.Disabled` for this to keep the existing behaviour. I'll follow this up with another PR that enables performance monitoring.

Note that I haven't worked with coroutines for a long time and I have never worked with `Flow` before, so I'd appreciate any feedback on what I might have missed or if there is a better way to achieve the changes I tried to make.


**To test:**
1. Throw an exception from `UnifiedAboutActivity.onCreate`
2. Run the app in `release` variant - I don't think it works for `debug` variant, but I am not 100%
3. Navigate to the About page and verify that the app crashes (Tap profile icon on the top right and tap on "About WordPress")
4. Check the latest events in Sentry and verify that the crash is reported correctly - [Here is the crash from my test](https://a8c.sentry.io/share/issue/efb8a23eb68f4d71b8f5eea2ad3e2ab8/).
5. Verify that your user information is showing up correctly in the Sentry event.

## Regression Notes
1. Potential unintended areas of impact
This might possibly break Sentry integration for the project

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I manually crashed the app and verified that the crash is reported to Sentry

3. What automated tests I added (or what prevented me from doing so)
N/A

---

@wzieba @ParaskP7 Since this PR depends on another PR, I am leaving it in draft state, but I was hoping that you could still review it so that I can build on it to add Sentry performance monitoring.

---

PR submission checklist:


UI Changes testing checklist:
